packages/ember-metal/lib/observer_set.js

```diff
@@ -28,28 +25,20 @@ export default class ObserverSet {
       this.added.set(object, keys);
     }
 
-    if (!keys.has(key)) {
-      this.queue.push({ object, key });
-      keys.add(key);
-    }
+    keys.add(key);
   }
 
-  flush() {
-    for (let i = 0; i < this.queue.length; i++) {
-      let { object, key } = this.queue[i];
-
-      if (object.isDestroying || object.isDestroyed) {
-        continue;
+  forEach(cb) {
+    this.added.forEach(function(keys, object) {
```

Comment:
You need to loop over `this.queue` not `this.added` otherwise the observers may be triggered in the wrong order

Reply:
ok got it

Reply:
Arrow functions too ;)